src/main/java/de/tum/in/www1/artemis/web/rest/UserResource.java

```diff
@@ -196,6 +197,31 @@ public ResponseEntity<UserDTO> updateUser(@Valid @RequestBody ManagedUserVM mana
         return ResponseEntity.ok().headers(HeaderUtil.createAlert(applicationName, "userManagement.updated", managedUserVM.getLogin())).body(new UserDTO(updatedUser));
     }
 
+    /**
+     * PUT ldap : Updates an existing User based on the info available in the LDAP server.
+     *
+     * @param login of the user to update
+     * @return the ResponseEntity with status 200 (OK) and with body the updated user
+     */
+    @PutMapping("users/{login:" + Constants.LOGIN_REGEX + "}/sync-ldap")
+    @PreAuthorize("hasRole('ADMIN')")
+    @Profile("ldap")
+    public ResponseEntity<UserDTO> syncUserViaLdap(@PathVariable String login) {
+        log.debug("REST request to update ldap information User : {}", login);
+
+        var existingUser = userRepository.findOneWithGroupsAndAuthoritiesByLogin(login);
```

Comment:
By using the `userId` above, you can use the method `findByIdWithGroupsAndAuthoritiesElseThrow` here which simplifies the code